### PR TITLE
Add config

### PR DESCRIPTION
**Pull request by bk201-**

### CODE REVIEW

**Comment by bk201-** [comment]
@hy0ug0 Please take a look this PR. I tried to add more flexible config which can solve most problems and it covers all previous PRs

File: src/config.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    debug(`Loaded config: ${JSON.stringify(config)}`);
+
+    if (config && !config.isEmpty) {
+      const result = { ...defaultConfig, ...config.config };
+      debug(`Used config: ${JSON.stringify(result)}`);
+      return result;
+    }
+  } catch {
+    // ignore
```
Comment (hy0ug0) [inline src/config.ts]: I would suggest to log something in case of an error is catched

**Comment by bk201-** [comment]
@MrRefactoring

**Comment by hy0ug0** [comment]
Looks very nice, good job man.

**Comment by bk201-** [comment]
> Looks very nice, good job man.
Thanks! 

I'll add a description into README tomorrow and will merge PR.

**Comment by hy0ug0** [comment]
Sounds good to me, I gonna close my PR since it's resolved by yours

**Pull request merged**